packages/errors/src/serialization/error.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { JsonObject } from '@backstage/config';
+import {
+  deserializeError as deserializeErrorInternal,
+  serializeError as serializeErrorInternal,
+} from 'serialize-error';
+
+/**
+ * The serialized form of an Error.
+ */
+export type SerializedError = JsonObject & {
+  /** The name of the exception that was thrown */
+  name: string;
+  /** The message of the exception that was thrown */
+  message: string;
+  /** A stringified stack trace; may not be present */
+  stack?: string;
+  /** A custom code (not necessarily the same as an HTTP response code); may not be present */
+  code?: string;
+};
+
+/**
+ * Serializes an error object to a JSON friendly form.
+ *
+ * @param error The error
+ * @param options.includeStackTraces: Include stack trace in the output (default false)
+ */
+export function serializeError(
+  error: Error,
+  options?: { includeStack?: boolean },
+): SerializedError {
+  const serialized = serializeErrorInternal(error);
```

Comment:
Hmm, bit uneasy blindly serializing all fields, both because API becomes a bit broad, but also because I could see some error objects in the backend having things that we might not necessarily want to share, like file paths and IPs. Might be better to deal with that where applicable though, and not blindly set any kind of error as an upstream cause

Reply:
So what are you saying, keep this as is? :)